litedict.py

```diff
@@ -113,11 +124,15 @@ def __repr__(self):
         return f"{type(self).__name__}(Connection={self.conn!r}, items={len(self)})"
 
     def glob(self, pat: str):
-        c = self.conn.execute("SELECT value FROM Dict WHERE Key GLOB ?", (pat,))
-        rows = c.fetchall()
-        if rows is None:
+        c = self.conn.execute("SELECT Key FROM Dict WHERE Key GLOB ?", (pat,))
+        keys = c.fetchall()
+        if keys is None:
             raise KeyError(pat)
-        return [self.decoder(x[0]) for x in rows]
+        return [self[key[0]] for key in keys]
+    
+    def sync(self):
+        for key,val in self._writeback.items():
+            self[key] = val
```

Comment:
This loop modifies the `self._writeback` dictionary while iterating over it, which is not desirable. Calling `self[key]` will call `__setitem__`, which is now doing `self._writeback[key] = value`.

I guess the most straightforward solution is splitting `__setitem__` into 2 methods, one modifies the writeback and one `INSERT`'ing into the database. Then the `.sync()` function would only call the method that writes to the SQLite database.

Reply:
I thought about this while I was doing it and decided it was okay since it doesn't (a) change the keys or (b) modify any values before iteration. I think that is okay, right? If not, I'll update